cpp/ql/src/semmle/code/cpp/security/Encryption.qll

```diff
@@ -42,17 +53,20 @@ string algorithmWhitelist() {
   result = "ECIES"
 }
 
-/** A regex for matching strings that look like they contain a whitelisted algorithm */
+/**
+ * Returns a regular expression for matching strings that look like they
+ * contain an algorithm that is known to be secure.
+ */
 string algorithmWhitelistRegex() {
-  // The implementation of this is a duplicate of algorithmBlacklistRegex, as it isn't
-  // possible to have string -> string functions at the moment
+  // The implementation of this is a duplicate of algorithmBlacklistRegex, as
+  // it isn't possible to have string -> string functions at the moment
```

Comment:
Out of curiosity: Does this comment predate `bindingset` annotations? I guess it should be possible to have something like:
```codeql
bindingset[regexMiddle]
string algorithmRegex(string regexMiddle) {
 // ...
}
```

Reply:
Possibly, but I think we had `pragma[inline]` when this was written.

I've now deleted this part of the comment.  I'm not convinced complicating the solution with a separate string -> string function is actually worthwhile in this case, since there are only two predicates that would use it.

Reply:
It turns out Java does have a string -> string predicate.  This was probably done at a later date than both libraries were originally created.